File: backend/src/main/java/com/aivideo/canvas/entity/Project.java
package com.aivideo.canvas.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity @Table(name = "projects") @Data
public class Project {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long userId;
    private String name;
    @Column(columnDefinition = "TEXT")
    private String description;
    private String coverUrl;
    private Long latestVersionId;
    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();
}

File: backend/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel


class ProjectCreateRequest(BaseModel):
    name: str
    description: str = ""


class ProjectUpdateRequest(BaseModel):
    name: str
    description: str = ""
    cover_url: str = ""


class ProjectData(BaseModel):
    id: int
    name: str
    description: str
    cover_url: str
    latest_version_id: int | None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
